File: app/src/main/java/com/tevinjeffrey/rutgersct/dagger/ActivityBindingModule.java
package com.tevinjeffrey.rutgersct.dagger;

import android.app.Activity;
import com.tevinjeffrey.rutgersct.ui.MainActivity;
import com.tevinjeffrey.rutgersct.ui.MainActivitySubcomponent;
import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

@Module(subcomponents = MainActivitySubcomponent.class)
public abstract class ActivityBindingModule {
  @Binds
  @IntoMap
  @ActivityKey(MainActivity.class)
  abstract AndroidInjector.Factory<? extends Activity>
  bindMainActivityInjectorFactory(MainActivitySubcomponent.Builder builder);
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/dagger/ActivityBindingModule.kt
package com.tevinjeffrey.rutgersct.dagger

import android.app.Activity

import com.tevinjeffrey.rutgersct.ui.MainActivity

import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap

@Module(subcomponents = [MainActivitySubcomponent::class])
abstract class ActivityBindingModule {
  @Binds
  @IntoMap
  @ActivityKey(MainActivity::class)
  abstract fun bindMainActivityInjectorFactory(
      builder: MainActivitySubcomponent.Builder): AndroidInjector.Factory<out Activity>
}